File: javascript/stations.js
// Find US weather stations near a city.
const BASE_URL = "https://api.usclimatedata.com/v1";
const API_KEY = process.env.USCLIMATEDATA_API_KEY;

const url = new URL(`${BASE_URL}/stations`);
url.searchParams.set("city", "New York");
url.searchParams.set("limit", "5");

const response = await fetch(url, { headers: { "X-API-Key": API_KEY } });
if (!response.ok) {
  throw new Error(`Request failed: ${response.status}`);
}
console.log(await response.json());

File: python/stations.py
"""Find US weather stations near a city."""
import os

import requests

BASE_URL = "https://api.usclimatedata.com/api/v1"
API_KEY = os.environ["USCLIMATEDATA_API_KEY"]

response = requests.get(
    f"{BASE_URL}/stations",
    headers={"X-API-Key": API_KEY},
    params={"city": "New York", "limit": 5},
    timeout=10,
)
response.raise_for_status()
print(response.json())
